parser/parser.mly

```diff
@@ -34,6 +28,20 @@
         | [t] -> t
         | f::a1::args -> PreApp (f,a1,args)
 
+    let mk_prelude lc md = mk_entry(Prelude(lc,md))
+
+    let mk_declaration lc id st t = mk_entry(Declaration(lc,id,st,t))
+
+    let mk_definition lc id pty te = mk_entry(Definition(lc,id,false,pty,te))
+
+    let mk_opaque lc id pty te = mk_entry(Definition(lc,id,true,pty,te))
+
+    let mk_rules rs = mk_entry(Rules(rs))
+
+    let mk_command lc cmd = mk_entry(Command(lc,cmd))
+
+    let mk_ending = mk_entry(Ending)
```

Comment:
Isn't this a fully applied function call which is therefore going to be evaluated at functor's instantiation and not when encountering EoF ?
If so I would suggest `let mk_ending () = mk_entry(Ending)` instead.